### PR TITLE
Refactor phi gc

### PR DESCRIPTION
**Pull request by mathause**
- [x] Tests added
 - [x] Passes `isort . && black . && flake8`
 - [x] Fully documented, including `CHANGELOG.rst`

After merging #59 I did a small refactoring of the `load_phi_gc` functionality. 

- vectorized `gaspari_cohn`
- Added `calc_geodist_exact` to it's own function
  - Used `geographiclib.geodesic.Geodesic` which is slightly faster than `geopy.distance` (geopy calls geographiclib under the hood)
  - only calculate the upper half of the matrix
- added some unittests for `gaspari_cohn` and `calc_geodist_exact`

TODOs (in another PR):
- This should not live inside `load_constant_files.py`. But I am not sure how to name the file: `distance_utils.py`? `regularization.py`? Something better?
- I would still like to do #62.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,10 @@ Documentation
 Internal Changes
 ^^^^^^^^^^^^^^^^
 
+ - Refactor and speed up of the gaspari_cohn function and the calculation of the great
```
Comment (mathause) [inline CHANGELOG.rst:30]:
```suggestion
 - Refactor and speed up of the Gaspari-Cohn function and the calculation of the great
```

**Comment by mathause** [comment]
Anyone planing on reviewing this? With #59 in place this should be a safe change so I will merge soonish if I don't hear back from you.

File: mesmer/io/load_constant_files.py [hunk 2]
```diff
@@ -37,29 +36,62 @@ def gaspari_cohn(r):
...
+        + 5 / 8 * r_s ** 3
+        - 1 / 2 * r_s ** 4
+        + 1 / 12 * r_s ** 5
+        - 2 / (3 * r_s)
+    )
+
+    return y.reshape(shape)
+
+
+def calc_geodist_exact(lon, lat):
```
Comment (mathause) [inline mesmer/io/load_constant_files.py:74]:
```suggestion
def calc_geodist_exact(lon, lat):
    """exact great circle distance based on WSG 84

    Parameters
    ----------
    lon : array-like
        1D array of longitudes
    lat : array-like
        1D array of latitudes

    Returns
    -------
    geodist : np.array
        2D array of great circle distances.
    """
```

**Review by leabeusch — APPROVED** [review]
Finally had a (too?) quick look at this. Looks all very nice at first glance (also yayy for now actually tests being added as parts of PRs. :thumbsup:). I didn't go into the code in detail (I assume / hope all is fine with it since it doesn't break your `phi_gc` test), but I do have a remark about how to name it: I would not move this to a script called `regularization.py`. The regularization itself happens in `train_lv()` (in Lines 349 & 382). Potentially `regularization_utils.py`? Although I personally wouldn't expect distance calculations in a script like that at all (but instead statistical methods for L1 / L2 regularizations etc).... so maybe just keep the `gaspari_cohn()` & `load_phi_gc()` part in the `regularization_utils.py` script and add yet another script just with the distance calculations called `distance_utils.py`? Then you could also put your great circle approx. there once you have it. Not sure if this is starting to be too many individual files then. But then again, usually you guys do encourage me to separate things more. :smile:

**Comment by mathause** [comment]
Thanks for the feedback. I think I would probably move `calc_geodist_exact` to `mesmer.utils.geodist`. `gaspari_cohn` could go to `mesmer.utils.statistics` (or `mesmer.utils.kernels` (is it a kernel?)). But indeed one function per file is a bit overkill. So both could go to `mesmer.utils.computation`? (Naming is hard!)

**Pull request merged**